pkg/cmd/admin/router/router.go

```diff
@@ -457,10 +464,17 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 		return fmt.Errorf("unable to parse --ports: %v", err)
 	}
 
-	// For the host networking case, ensure the ports match. Otherwise, remove host ports
-	for i := 0; i < len(ports); i++ {
-		if cfg.HostNetwork && ports[i].HostPort != 0 && ports[i].ContainerPort != ports[i].HostPort {
-			return fmt.Errorf("when using host networking mode, container port %d and host port %d must be equal", ports[i].ContainerPort, ports[i].HostPort)
+	// HostNetwork overrides HostPorts
+	if cfg.HostNetwork {
+		cfg.HostPorts = false
```

Comment:
We can't, we'd need to refactor to options.  Net new flag, I'm ok with that
being a bug.

On Thu, Mar 31, 2016 at 12:53 AM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/cmd/admin/router/router.go
> <https://github.com/openshift/origin/pull/8300#discussion_r58001126>:
>
> > @@ -457,10 +464,17 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
> >  		return fmt.Errorf("unable to parse --ports: %v", err)
> >  	}
> >
> > -	// For the host networking case, ensure the ports match. Otherwise, remove host ports
> > -	for i := 0; i < len(ports); i++ {
> > -		if cfg.HostNetwork && ports[i].HostPort != 0 && ports[i].ContainerPort != ports[i].HostPort {
> > -			return fmt.Errorf("when using host networking mode, container port %d and host port %d must be equal", ports[i].ContainerPort, ports[i].HostPort)
> > +	// HostNetwork overrides HostPorts
> > +	if cfg.HostNetwork {
> > +		cfg.HostPorts = false
>
> unfortunate we can't tell here if the user explicitly passed --host-network
> --host-ports (which should probably be a validation error?) or if we're
> just adjusting the default
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8300/files/1fd121b487f37d90f41daca4fe6d3cd9b51896c6#r58001126>
>